chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/AddCuresUpdateToListingsJob.java

```diff
@@ -0,0 +1,127 @@
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.manager.ActivityManager;
+import gov.healthit.chpl.manager.CertifiedProductDetailsManager;
+import gov.healthit.chpl.service.CuresUpdateService;
+
+public class AddCuresUpdateToListingsJob extends QuartzJob {
+    private static final Logger LOGGER = LogManager.getLogger("addCuresUpdateToListingsJobLogger");
+
+    @Autowired
+    private CuresUpdateService curesUpdateService;
+
+    @Autowired
+    private CertifiedProductDetailsManager certifiedProductDetailsManager;
+
+    @Autowired
+    private CertifiedProductDAO certifiedProductDAO;
+
+    @Autowired
+    private SpecialProperties specialProperties;
+
+    @Autowired
+    private CuresUpdateEventDAO curesUpdateDao;
+
+    @Autowired
+    private ActivityManager activityManager;
+
+    @Autowired
+    private JpaTransactionManager txManager;
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        LOGGER.info("********* Starting the Update Listing To Cures Updated job. *********");
+
+        getAll2015Listings().stream()
+                .forEach(dto -> processListing(dto));
+
+        LOGGER.info("********* Completed the Update Listing To Cures Updated  job. *********");
+
+    }
+
+    private void processListing(CertifiedProductDetailsDTO cp) {
+        try {
+            LOGGER.info("Retrieving listing information for  (" + cp.getId() + ") " + cp.getChplProductNumber());
+            CertifiedProductSearchDetails listing = certifiedProductDetailsManager.getCertifiedProductDetails(cp.getId());
+            if (curesUpdateService.isCuresUpdate(listing)) {
+                LOGGER.info("************************** Updating " + cp.getId() + " as Cures Updated **************************");
+                updateListingAsCuresUpdated(listing);
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+    }
+
+    private void updateListingAsCuresUpdated(CertifiedProductSearchDetails origListing)
+            throws EntityCreationException, EntityRetrievalException, JsonProcessingException {
+
+        TransactionTemplate txTemplate = new TransactionTemplate(txManager);
+        txTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
+        txTemplate.execute(new TransactionCallbackWithoutResult() {
+
+            @Override
+            // can't use the manager method because we don't want to record activity
+            // so wrapping the dao call in a transaction
+            protected void doInTransactionWithoutResult(TransactionStatus status) {
+                try {
+                    addCuresUpdateEvent(origListing.getId(), true, specialProperties.getEffectiveRuleDate());
+                    addCuresUpdateActivity(origListing,
```

Comment:
This is a little confusing because the comment above specifically says you are doing this so that you don't add activity but then you are calling a method that says it's adding activity.